### ISSUE TITLE
Keep and reuse downloaded installers

### ISSUE THREAD

**Issue by maln0ir** [lutris/lutris#2391]
**Describe the bug**

1. Lutris removes downloaded game files even if installation script fails and user explicitly said to not remove files. A popup window shows up where I can uncheck option to remove installer, but files are removed anyway.
2. Lutris redownloads installer files even if it exists and is correctly downloaded

Some games are huge and it is really irritating when after a few hours of downloading installation fails.

**Expected behavior**

1. Do not remove downloaded installers if user wants to keep them
2. Use installers without downloading them if they are available in cache.

**Steps to reproduce**

Try to install Witcher 2: Assasins of Kings Enhanced Edition from GOG.com

**Lutris debugging output**

Log from installation of Witcher 2 from GOG.com

```INFO     2019-10-08 18:47:24,141 [application.do_command_line:225]:Running Lutris 0.5.3
INFO     2019-10-08 18:47:24,141 [startup.check_driver:60]:Using X.Org
INFO     2019-10-08 18:47:24,141 [startup.check_driver:65]:Running Mesa driver 19.2.0 on Radeon1-1-zen, LLVM 8.0.1) (0x67df)
INFO     2019-10-08 18:47:24,141 [startup.check_driver:75]:GPU: 1002:67DF 1462:341B using amdg
INFO     2019-10-08 18:47:24,186 [startup.check_vulkan:129]:Vulkan is supported
INFO     2019-10-08 18:47:24,187 [dxvk.get_dxvk_versions:21]:Updating DXVK versions
DEBUG    2019-10-08 18:47:24,373 [http.get:65]:GET https://lutris.net/api/runtime
DEBUG    2019-10-08 18:47:24,378 [store.get_missing_media:198]:Requesting missing icons from A
DEBUG    2019-10-08 18:47:24,379 [http.get:65]:GET https://lutris.net/api/games
DEBUG    2019-10-08 18:47:24,789 [api.get_game_api_page:132]:Loaded 2 games from page 1
INFO     2019-10-08 18:47:25,236 [dxvk.get_dxvk_versions:21]:Updating D9VK versions
DEBUG    2019-10-08 18:48:25,590 [interpreter.fetch_script:54]:Fetching installer https://lutris.net/api/installers/the-witcher-2-assassins-of-kings-enhanced-edition
DEBUG    2019-10-08 18:48:25,591 [http.get:65]:GET https://lutris.net/api/installers/the-witcher-2-assassins-of-kings-enhanced-edition
DEBUG    2019-10-08 18:48:30,828 [display._get_vidmodes:15]:Retrieving video modes from XrandR
DEBUG    2019-10-08 18:48:32,642 [interpreter.check_runner_install:440]:Required runners: [<lutris.runners.linux.linux object at 0x7f139db3e390>]
DEBUG    2019-10-08 18:48:32,645 [http.get:65]:GET https://auth.gog.com/token?[REDACTED]
DEBUG    2019-10-08 18:48:33,164 [gog.load_cookies:111]:Loading cookies from /home/ill/.cache/lutris/.gog.auth
DEBUG    2019-10-08 18:48:33,167 [http.get:65]:GET https://embed.gog.com/userData.json
DEBUG    2019-10-08 18:48:33,630 [http.get:65]:GET https://embed.gog.com/userData.json
DEBUG    2019-10-08 18:48:33,927 [gog.load_cookies:111]:Loading cookies from /home/ill/.cache/lutris/.gog.auth
DEBUG    2019-10-08 18:48:33,929 [http.get:65]:GET https://embed.gog.com/userData.json
DEBUG    2019-10-08 18:48:34,227 [http.get:65]:GET https://embed.gog.com/userData.json
INFO     2019-10-08 18:48:34,517 [gog.get_game_details:204]:Getting game details for 1207658930
DEBUG    2019-10-08 18:48:34,517 [gog.load_cookies:111]:Loading cookies from /home/ill/.cache/lutris/.gog.auth
DEBUG    2019-10-08 18:48:34,519 [http.get:65]:GET https://api.gog.com/products/1207658930?expand=downloads
DEBUG    2019-10-08 18:48:34,833 [http.get:65]:GET https://api.gog.com/products/1207658930?expand=downloads
INFO     2019-10-08 18:48:35,057 [gog.get_download_info:210]:Getting download info for http://api.gog.com/products/1207658930/downlink/installer/en3installer0
DEBUG    2019-10-08 18:48:35,057 [gog.load_cookies:111]:Loading cookies from /home/ill/.cache/lutris/.gog.auth
DEBUG    2019-10-08 18:48:35,058 [http.get:65]:GET http://api.gog.com/products/1207658930/downlink/installer/en3installer0
DEBUG    2019-10-08 18:48:35,254 [http.get:65]:GET http://api.gog.com/products/1207658930/downlink/installer/en3installer0
INFO     2019-10-08 18:48:35,429 [interpreter.get_gog_download_links:977]:Adding https://cdn.gog.com/secure/linux_offlines/1207658930/1207658930/664653772/1462/the_witcher_2_assassins_of_kings_enhanced_edition_en_release_3_20150306204412_20992.sh.xml?[REDACTED] to download links
INFO     2019-10-08 18:48:35,429 [interpreter.get_gog_download_links:977]:Adding https://cdn.gog.com/secure/linux_offlines/1207658930/1207658930/664653772/1462/the_witcher_2_assassins_of_kings_enhanced_edition_en_release_3_20150306204412_20992.sh?[REDACTED] to download links
DEBUG    2019-10-08 18:48:35,430 [interpreter.swap_gog_game_files:318]:Removing file goginstaller
DEBUG    2019-10-08 18:48:35,430 [interpreter.swap_gog_game_files:337]:Adding GOG file the_witcher_2_assassins_of_kings_enhanced_edition_en_release_3_20150306204412_20992.sh.xml as gog_file_0
DEBUG    2019-10-08 18:48:35,430 [interpreter.swap_gog_game_files:337]:Adding GOG file the_witcher_2_assassins_of_kings_enhanced_edition_en_release_3_20150306204412_20992.sh as goginstaller
INFO     2019-10-08 18:48:35,431 [interpreter.iter_game_files:374]:Downloading file 1 of 2
DEBUG    2019-10-08 18:48:35,431 [installer_file.get_download_info:96]:Downloading [gog_file_0]: https://cdn.gog.com/secure/linux_offlines/1207658930/1207658930/664653772/1462/the_witcher_2_assassins_of_kings_enhanced_edition_en_release_3_20150306204412_20992.sh.xml?[REDACTED] to /home/ill/.cache/lutris/installer/the-witcher-2-assassins-of-kings-enhanced-edition/gog/the_witcher_2_assassins_of_kings_enhanced_edition_en_release_3_20150306204412_20992.sh.xml
DEBUG    2019-10-08 18:48:35,434 [installerwindow.start_download:319]:Downloading https://cdn.gog.com/secure/linux_offlines/1207658930/1207658930/664653772/1462/the_witcher_2_assassins_of_kings_enhanced_edition_en_release_3_20150306204412_20992.sh.xml?[REDACTED] to /home/ill/.cache/lutris/installer/the-witcher-2-assassins-of-kings-enhanced-edition/gog/the_witcher_2_assassins_of_kings_enhanced_edition_en_release_3_20150306204412_20992.sh.xml
DEBUG    2019-10-08 18:48:35,436 [downloader.start:55]:Starting download of:
 https://cdn.gog.com/secure/linux_offlines/1207658930/1207658930/664653772/1462/the_witcher_2_assassins_of_kings_enhanced_edition_en_release_3_20150306204412_20992.sh.xml?[REDACTED]
DEBUG    2019-10-08 18:48:35,934 [downloader.on_done:97]:Finished downloading https://cdn.gog.com/secure/linux_offlines/1207658930/1207658930/664653772/1462/the_witcher_2_assassins_of_kings_enhanced_edition_en_release_3_20150306204412_20992.sh.xml?[REDACTED]
INFO     2019-10-08 18:48:35,937 [interpreter.iter_game_files:374]:Downloading file 2 of 2
DEBUG    2019-10-08 18:48:35,938 [installer_file.get_download_info:96]:Downloading [goginstaller]: https://cdn.gog.com/secure/linux_offlines/1207658930/1207658930/664653772/1462/the_witcher_2_assassins_of_kings_enhanced_edition_en_release_3_20150306204412_20992.sh?[REDACTED] to /home/ill/.cache/lutris/installer/the-witcher-2-assassins-of-kings-enhanced-edition/gog/the_witcher_2_assassins_of_kings_enhanced_edition_en_release_3_20150306204412_20992.sh
DEBUG    2019-10-08 18:48:35,939 [installerwindow.start_download:319]:Downloading https://cdn.gog.com/secure/linux_offlines/1207658930/1207658930/664653772/1462/the_witcher_2_assassins_of_kings_enhanced_edition_en_release_3_20150306204412_20992.sh?[REDACTED] to /home/ill/.cache/lutris/installer/the-witcher-2-assassins-of-kings-enhanced-edition/gog/the_witcher_2_assassins_of_kings_enhanced_edition_en_release_3_20150306204412_20992.sh
DEBUG    2019-10-08 18:48:35,940 [downloader.start:55]:Starting download of:
 https://cdn.gog.com/secure/linux_offlines/1207658930/1207658930/664653772/1462/the_witcher_2_assassins_of_kings_enhanced_edition_en_release_3_20150306204412_20992.sh?[REDACTED]
DEBUG    2019-10-08 19:33:24,075 [http.get:65]:GET https://auth.gog.com/token?[REDACTED]
DEBUG    2019-10-08 19:33:25,302 [gog.load_cookies:111]:Loading cookies from /home/ill/.cache/lutris/.gog.auth
DEBUG    2019-10-08 19:33:25,304 [http.get:65]:GET https://embed.gog.com/userData.json
DEBUG    2019-10-08 19:33:25,744 [http.get:65]:GET https://embed.gog.com/userData.json
DEBUG    2019-10-08 19:33:26,068 [gog.load_cookies:111]:Loading cookies from /home/ill/.cache/lutris/.gog.auth
DEBUG    2019-10-08 19:33:26,069 [http.get:65]:GET https://embed.gog.com/userData.json
DEBUG    2019-10-08 19:33:26,438 [http.get:65]:GET https://embed.gog.com/userData.json
DEBUG    2019-10-08 19:33:35,128 [gog.load_cookies:111]:Loading cookies from /home/ill/.cache/lutris/.gog.auth
DEBUG    2019-10-08 19:33:35,131 [http.get:65]:GET https://embed.gog.com/userData.json
DEBUG    2019-10-08 19:33:35,493 [http.get:65]:GET https://embed.gog.com/userData.json
DEBUG    2019-10-08 19:33:35,824 [gog.get_library:175]:Returning cached GOG library
DEBUG    2019-10-08 19:33:49,082 [gog.load_cookies:111]:Loading cookies from /home/ill/.cache/lutris/.gog.auth
DEBUG    2019-10-08 19:33:49,084 [http.get:65]:GET https://embed.gog.com/userData.json
DEBUG    2019-10-08 19:33:49,610 [http.get:65]:GET https://embed.gog.com/userData.json
DEBUG    2019-10-08 19:33:49,949 [gog.load_cookies:111]:Loading cookies from /home/ill/.cache/lutris/.gog.auth
DEBUG    2019-10-08 19:33:49,950 [http.get:65]:GET https://embed.gog.com/account/getFilteredProducts?mediaType=1&page=1
DEBUG    2019-10-08 21:19:08,141 [downloader.on_done:97]:Finished downloading https://cdn.gog.com/secure/linux_offlines/1207658930/1207658930/664653772/1462/the_witcher_2_assassins_of_kings_enhanced_edition_en_release_3_20150306204412_20992.sh?[REDACTED]
DEBUG    2019-10-08 21:19:08,249 [interpreter._iter_commands:564]:Installer command: {'extract': {'file': 'goginstaller', 'dst': '$CACHE/GOG', 'format': 'zip'}}
DEBUG    2019-10-08 21:19:08,250 [commands.extract:174]:Extracting the_witcher_2_assassins_of_kings_enhanced_edition_en_release_3_20150306204412_20992.sh
DEBUG    2019-10-08 21:19:08,250 [commands.extract:178]:extracting file /home/ill/.cache/lutris/installer/the-witcher-2-assassins-of-kings-enhanced-edition/gog/the_witcher_2_assassins_of_kings_enhanced_edition_en_release_3_20150306204412_20992.sh to /home/ill/.cache/lutris/installer/the-witcher-2-assassins-of-kings-enhanced-edition/GOG
DEBUG    2019-10-08 21:19:08,262 [extract.extract_archive:69]:Extracting /home/ill/.cache/lutris/installer/the-witcher-2-assassins-of-kings-enhanced-edition/gog/the_witcher_2_assassins_of_kings_enhanced_edition_en_release_3_20150306204412_20992.sh to /home/ill/.cache/lutris/installer/the-witcher-2-assassins-of-kings-enhanced-edition/GOG

7-Zip [64] 16.02 : Copyright (c) 1999-2016 Igor Pavlov : 2016-05-21
p7zip Version 16.02 (locale=pl_PL.UTF-8,Utf16=on,HugeFiles=on,64 bits,8 CPUs x64)

Scanning the drive for archives:
1 file, 21185900162 bytes (20 GiB)                                       
Extracting archive: /home/ill/.cache/lutris/installer/the-witcher-2-assassins-of-kings-enhanced-edition/gog/the_witcher_2_assassins_of_kings_enhanced_edition_en_release_3_20150306204412_20992.sh
--
Path = /home/ill/.cache/lutris/installer/the-witcher-2-assassins-of-kings-enhanced-edition/gog/the_witcher_2_assassins_of_kings_enhanced_edition_en_release_3_20150306204412_20992.sh
Type = zip
Offset = 973365
Physical Size = 21184926797
64-bit = +

Everything is Ok                                                             
Folders: 73
Files: 1010
Size:       24141570179
Compressed: 21185900162
DEBUG    2019-10-08 21:22:34,237 [system.remove_folder:197]:Removing folder /home/ill/.cache/lutris/installer/the-witcher-2-assassins-of-kings-enhanced-edition/GOG/.extract-ed9b750a
DEBUG    2019-10-08 21:22:34,238 [extract.extract_archive:152]:Finished extracting /home/ill/.cache/lutris/installer/the-witcher-2-assassins-of-kings-enhanced-edition/gog/the_witcher_2_assassins_of_kings_enhanced_edition_en_release_3_20150306204412_20992.sh to /home/ill/.cache/lutris/installer/the-witcher-2-assassins-of-kings-enhanced-edition/GOG
DEBUG    2019-10-08 21:22:34,281 [interpreter._iter_commands:564]:Installer command: {'merge': {'src': '$CACHE/GOG/noarch/', 'dst': '$GAMEDIR'}}
DEBUG    2019-10-08 21:22:34,284 [commands.merge:259]:Merging /home/ill/.cache/lutris/installer/the-witcher-2-assassins-of-kings-enhanced-edition/GOG/noarch into /home/ill/games/the-witcher-2-assassins-of-kings-enhanced-edition
ERROR    2019-10-08 21:22:34,284 [errors.__init__:15]:Source does not exist: /home/ill/.cache/lutris/installer/the-witcher-2-assassins-of-kings-enhanced-edition/GOG/noarch
{'src': '$CACHE/GOG/noarch/', 'dst': '$GAMEDIR'}
ERROR    2019-10-08 21:22:34,284 [jobs.target:32]:Error while completing task <bound method CommandsMixin.merge of <lutris.installer.interpreter.ScriptInterpreter object at 0x7f13a45b5790>>: Source does not exist: /home/ill/.cache/lutris/installer/the-witcher-2-assassins-of-kings-enhanced-edition/GOG/noarch
{'src': '$CACHE/GOG/noarch/', 'dst': '$GAMEDIR'}
<class 'lutris.installer.errors.ScriptingError'> Source does not exist: /home/ill/.cache/lutris/installer/the-witcher-2-assassins-of-kings-enhanced-edition/GOG/noarch
{'src': '$CACHE/GOG/noarch/', 'dst': '$GAMEDIR'}
  File "/usr/lib/python3.7/site-packages/lutris/util/jobs.py", line 30, in target
    result = self.function(*args, **kwargs)
  File "/usr/lib/python3.7/site-packages/lutris/installer/commands.py", line 261, in merge
    raise ScriptingError("Source does not exist: %s" % src, params)
DEBUG    2019-10-08 21:23:26,555 [interpreter.revert:732]:Install cancelled
DEBUG    2019-10-08 21:23:26,555 [system.remove_folder:197]:Removing folder /home/ill/.cache/lutris/installer/the-witcher-2-assassins-of-kings-enhanced-edition
```

**System information**

OS: Arch Linux
Lutris version: 0.5.3.r25.ga136c4cd-1

**Screenshots**

**Comment by LeXofLeviafan** [comment]
…Are you sure you set up the client correctly? You need to have an accessible full path in the "Cache path" option (something like `/home/<username>/.cache/lutris/downloads`) for this feature to work.

**Comment by maln0ir** [comment]
I didn't know this is a separate feature. I assumed `~/.cache/lutris/installer` is the cache directory where _all_ files are downloaded.

Is it really needed to have _two_ caches that behave differently?

**Comment by LeXofLeviafan** [comment]
Downloaded installers are treated as temporary files by default. The "Cache path" option is intended specifically to override that default behaviour (though it won't work with the `~` shorthand); this is clearly stated right under the option in the preferences dialog.

BTW, the path I provided as example coincides with the one used for default downloads (which is why I set it up in my client) but it doesn't have to (for example if you keep your `.cache` in tmpfs).
